src/monocle_apptrace/metamodel/spans/span_example.json

```diff
@@ -0,0 +1,140 @@
+            }
+        }
+
+    ],
+    "links": [],
+    "resource": {
+        "attributes": {
+            "service.name": "coffee-bot"
+        },
+        "schema_url": ""
+    }
+},
+{
+    "name": "llamaindex.openai",
+    "context": {
+        "trace_id": "0x93cd0bf865b3ffcc3cf9c075dc3e3797",
+        "span_id": "0x8b6363e1937a4d7b",
+        "trace_state": "[]"
+    },
+    "kind": "SpanKind.CLIENT",
+    "parent_id": "0x7a63d63e42ccac60",
+    "start_time": "2024-09-09T14:38:45.622174Z",
+    "end_time": "2024-09-09T14:38:46.514120Z",
+    "status": {
+        "status_code": "OK"
+    },
+    "attributes": {
+        "span.type": "Inference",
+        "entity.count": 2,
+        "entity.1.name": "AzureOpenAI",
+        "entity.1.type": "Inference.Azure_oai",
+        "entity.1.provider_name": "openai.azure.com",
+        "entity.1.deployment": "kshitiz-gpt",
+        "entity.1.inference_endpoint": "https://okahu-openai-dev.openai.azure.com/",
+
+        "entity.2.name": "gpt-35-turbo",
+        "entity.2.type": "Model.LLM",
+        "entity.2.model_name": "gpt-35-turbo"
+    },
+    "events": [
+        {
+            "name": "data.input",
+            "timestamp": "timestamp",
+            "attributes": {
+                "question": "What is an americano?",
+            }
+        },
+        {
+            "name": "data.output",
+            "timestamp": "timestamp",
+            "attributes": {
+                "response": "An americano is a type of coffee drink that is made by diluting an espresso shot with hot water at a 1:3 to 1:4 ratio, resulting in a drink that retains the complex flavors of espresso, but in a lighter way.",
+            }
+        },
+        {
+            "name": "metadata",
+            "timestamp": "timestamp",
+            "attributes": {
+                "temperature": 0.1,
+                "completion_tokens": 52,
+                "prompt_tokens": 233,
+                "total_tokens": 285
+            }
+        }
+    ],
+    "links": [],
+    "resource": {
+        "attributes": {
+            "service.name": "coffee-bot"
+        },
+        "schema_url": ""
+    }
+}
+{
+    "name": "llamaindex.query",
+    "context": {
+        "trace_id": "0x93cd0bf865b3ffcc3cf9c075dc3e3797",
+        "span_id": "0x7a63d63e42ccac60",
+        "trace_state": "[]"
+    },
+    "kind": "SpanKind.CLIENT",
+    "parent_id": null,
+    "start_time": "2024-09-09T14:38:45.236627Z",
+    "end_time": "2024-09-09T14:38:46.514442Z",
+    "status": {
+        "status_code": "OK"
+    },
+    "attributes": {
+        "span.type": "Workflow",
+        "entity.count": 1,
+        "entity.1.name": "coffee-bot",
+        "entity.1.type": "Workflow.LlamaIndex"
```

Comment:
if keys are not fixed then database queries can become more complex

eg to get details where workflow name is give workflow:
in bigquery we can say `WHERE
        JSON_EXTRACT_SCALAR(t1.ATTRIBUTES, '$.workflow_name') = 'coffee-bot'`

now we might have to do some regex check on attributes json column, maybe something like 

WHERE REGEXP_CONTAINS(TO_JSON_STRING(json_column), r'"entity\.\d+\.name":\s*"coffee-bot"')

select expression in query might also become complex

as example consider query to get distinct workflows:

```
SELECT DISTINCT 
        JSON_EXTRACT_SCALAR(attributes, '$.workflow_name') AS WORKFLOW,
        JSON_EXTRACT_SCALAR(attributes, '$.workflow_type') AS WORKFLOW_TYPE
    FROM
        `{0}.traces`
    WHERE
        JSON_EXTRACT_SCALAR(attributes, '$.workflow_name') IS NOT NULL
        AND START_TIME > TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL {1} DAY) 
```


cc @oi-raanne

Reply:
checking..